tests/features/typing_584_test.py

```diff
@@ -0,0 +1,99 @@
+import pytest
+
+from pyupgrade._data import Settings
+from pyupgrade._main import _fix_plugins
+
+
+@pytest.mark.parametrize(
+    ('s', 'version'),
+    (
+        pytest.param(
+            '{**a, **b}\n',
+            (3, 8),
+            id='<3.9',
+        ),
+        pytest.param(
+            '{"a": 0}\n',
+            (3, 9),
+            id='Dict without merge',
+        ),
+    ),
+)
+def test_fix_pep584_noop(s, version):
+    assert _fix_plugins(s, settings=Settings(min_version=version)) == s
+
+
+@pytest.mark.parametrize(
+    ('s', 'expected'),
+    (
+        pytest.param(
+            'x = {**a, **b}\n',
+
+            'x = a | b\n',
+
+            id='Simple dict rewrite',
+        ),
+        pytest.param(
+            'x = {**a}\n',
+
+            'x = a\n',
+
+            id='Merge of only one dict',
+        ),
```

Comment:
this one isn't correct -- the _before_ is a shallow copy and the after is a direct reference

Reply:
Thanks, have updated!

Is there any drawback to using `Token('DUMMY', '')` or is `del tokens[i]` preferable?

Reply:
ah so, the reason for those is if it were to process things in reverse, I think you can just `del` those in this case.

an aside: I just updated my repositories to replace "dummy" with "placeholder" (or other equivalent language) to make the code more inclusive

Reply:
I see - I was having issues here when using `del`. Not sure I understand why `del` works at all actually if it changes the number of tokens, but I'll make sense of it

>  I just updated my repositories to replace "dummy" with "placeholder" (or other equivalent language) to make the code more inclusive

Thanks for doing that 👍 

And thanks for your review/merge!